datenguide_python/output_transformer.py

```diff
@@ -70,30 +72,37 @@ def build_and_merge_data(
                 ]
                 new_cols = [x.replace("data.region.", "") for x in cols_temp3]
                 globals()["data{}".format(i)].columns = new_cols
-                print("no type fields")
-
-            # ensure that data contain 'id' to merge and add columns from temp_list
-            # if available in query_result...
             try:
                 globals()["data{}".format(i)]["id"] = whole_data_body["data.region.id"][
                     0
                 ]
             except KeyError:
-                globals()["data{}".format(i)]["id"] = metadata["id"]
+                print(col_list_whole_data_body[i] + " has no id")
+                pass
             try:
                 globals()["data{}".format(i)]["name"] = whole_data_body[
                     "data.region.name"
                 ][0]
             except KeyError:
+                print(col_list_whole_data_body[i] + " has no name")
                 pass
 
         data_out = globals()["data{}".format(0)]
-        for l in range(0, len(col_list_whole_data_body)):
-            data_out = data_out.merge(globals()["data{}".format(l)], how="outer")
+        for l in range(1, len(col_list_whole_data_body)):
+            try:
+                data_out = data_out.merge(globals()["data{}".format(l)], how="outer")
+            except MergeError:
+                print("no joint key to merge on")
+                data_out["fake_id"] = 1
```

Comment:
actually one might consider always introducing the *fake_id* and avoid the try/except as it does not influence the case when there are other columns to join on.